core/src/main/java/com/scalar/db/storage/multistorage/MultiStorageAdmin.java

```diff
@@ -200,6 +211,18 @@ public void importTable(String namespace, String table) throws ExecutionExceptio
     getAdmin(namespace, table).importTable(namespace, table);
   }
 
+  @Override
+  public DistributedStorageMetadata getDistributedStorageMetadata(String namespace)
+      throws ExecutionException {
+    // Get the storage metadata from the admin instance for the namespace
+    DistributedStorageMetadata storageMetadata =
+        getAdmin(namespace).getDistributedStorageMetadata(namespace);
+
+    // Replace the name of the metadata with the storage name in the multi-storage configuration
+    String storageName = namespaceStorageNameMap.getOrDefault(namespace, defaultStorageName);
+    return DistributedStorageMetadata.newBuilder(storageMetadata).name(storageName).build();
+  }
```

Comment:
In `MultiStorageAdmin.getDistributedStorageMetadata()`, we first get the storage metadata from the appropriate admin for the namespace, and we also need to replace the name of the metadata with the storage name in the multi-storage configuration.